cpp/src/knn/util.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "cuda_utils.h"
+
+#include "utils.h"
+#include <cuda_runtime.h>
+
+namespace ML {
```

Comment:
It's funny you ask this, because when I started working on this PR, I had originally put it in the prims. Ideally, this should have its own set of tests, though it is currently being tested by proxy through the kNN mg tests. If I move this into the prims, it will require an mg test in the prims, which we are keeping internal. One option could be a separate test build for spmg prims, I suppose.

Given my understanding of a long-term goal to integrate our Dask cuML OPG algorithms seamlessly into the user's environment, I am personally hoping to see the SPMG algorithms go away altogether. Taking this into account, we could also move this into a shared utility in the algorithms layer, which would still allow it to be tested in the "ml_mg" tests. For now, I pulled it out into a utility under kNN until we actually have another algorithm that wants to use it.

Reply:
Sorry for the delayed response! I totally see your point, probably I would think to move it as a utility in the algorithms layer might be the best option to avoid extra work but still give it visibility in case it could be useful for some developer, what do you think?